icaruscode/CRT/CRTUtils/CRTHitRecoAlg.h

```diff
@@ -196,4 +196,805 @@ class icarus::crt::CRTHitRecoAlg {
   }
 }; //class CRTHitRecoAlg
 
+inline bool sortbysec(const pair<int,ULong64_t> &a,
+              const pair<int,ULong64_t> &b)
+{
+    return (a.second < b.second);
+}
+
+inline ULong64_t GetMode(std::vector<std::pair<int, ULong64_t>> vector) {
+
+	sort(vector.begin(), vector.end(), sortbysec);
+
+	int modecounter = 0;
+	int isnewmodecounter = 0;
+	ULong64_t Mode = 0;
+	ULong64_t isnewMode = 0;
+	bool isFirst = true;
+	for (auto i : vector) {
+		if (!isFirst) {
+			if (i.second == Mode) modecounter++;
+			else if (i.second !=isnewMode) {
+				isnewMode = i.second;
+				isnewmodecounter = 1;
+			}
+			else if (i.second == isnewMode) {
+				isnewmodecounter++;
+				if (isnewmodecounter > modecounter) {
+					Mode = isnewMode;
+					modecounter = isnewmodecounter;
+				}
+			}
+		}
+		else {
+			isFirst = false;
+			Mode = i.second;
+			modecounter++;
+		}
+	}
+	return Mode;
+}
+
+typedef struct FEB_delay {
+	int HW_mac;
+	int SW_mac;
+	int SW_modID;
+	ULong64_t T0_delay;
+	ULong64_t T1_delay;
+} FEB_delay;
```

Comment:
The values are left uninitialised; this might make them zero'ed, but it's not so obvious: much better to be plain about it:
```suggestion
struct FEB_delay {
	int HW_mac = -1;
	int SW_mac = -1;
	int SW_modID = -1;
	ULong64_t T0_delay = 0; // [ns]
	ULong64_t T1_delay = 0; // [ns]
};
```
Also, delays might be better expressed signed, just in case, and a comment indicating which unit they are expected is helpful
(I have updated the definition style from C-like to C++-like too, "for free").

Reply:
Good point, I added this and overall restyled the whole LoadFEBMap function